nosnoc/nosnoc.py

```diff
@@ -1204,13 +1204,25 @@ def solve(self) -> dict:
     # TODO: move this to problem?
     def set(self, field: str, value: np.ndarray) -> None:
         """
-        :param field: in ["x", "p_global", "p_time_var"]
+        :param field: in ["x0", "x", "u", "p_global", "p_time_var"]
         :param value: np.ndarray: numerical value of appropriate size
         """
         prob = self.problem
         dims = prob.model.dims
-        if field == 'x':
+        if field == 'x0':
             prob.model.x0 = value
+        elif field == 'x':
+            if value.shape[0] == self.opts.N_stages + 1:
+                # Shape is equal to the number of stages
+                self.set("x0", value[1, :])
```

Comment:
I would not make x0 part of this, especially, now that it isn't an optimization variable.